### PR TITLE
Using Drupal as the CMS service

### PR DESCRIPTION
**Pull request by RoySegall**

### CODE REVIEW

**Comment by RoySegall** [comment]
Things to test:
1. Code sniffer
2. Testing the graphql
3. Testing the upload file controller
4. Test the view that send files to process
5. Unit tests for the services we created

**Comment by RoySegall** [comment]
Things to do:
* [ ] In the uploaded files handke ajax request
* [ ] Add Multiple file sending
* [ ] Read from the logs system

**Comment by RoySegall** [comment]
There are a couple of technical issues:
* [ ] When running the docker compose the install script is kind of jamming the container and prevent from the service to be ready
* [ ] When running the service with docker compose the Drupal installation is ready available in `http://localhost:8080/drupal/web` but when running the container as is Drupal is available from `http://localhost:8080`

@ItamarGronich Any ideas?

**Comment by RoySegall** [comment]
@ItamarGronich All the things are good. There are still some issues left with setting up the service. Let'a talk about it tomorrow.

File: docker-compose.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@ version: "3"
 volumes:
   db-data:
     external: false
+  my-db:
```
Comment (ItamarGronich) [inline docker-compose.yml]: external: false
```suggestion
  my-db:
  	external: false
```

File: docker-compose.yml [hunk 2]
```diff
@@ -39,6 +40,12 @@ services:
       - ./logs/django:/home/app/server/logs
     depends_on:
       - database
+      - cms
```
Comment (ItamarGronich) [inline docker-compose.yml]: remove

**Comment by RoySegall** [comment]
@ItamarGronich Can't figure it out. When running:
`docker-compose up cms`

I'm getting:
```
Removing open_pension_cms_1
open_pension_cms_db_1 is up-to-date
Recreating 6ee3ca860bfb_open_pension_cms_1 ... error

ERROR: for 6ee3ca860bfb_open_pension_cms_1  Cannot start service cms: OCI runtime create failed: container_linux.go:344: starting container process caused "exec: \"/var/www/html/entrypoint.sh\": stat /var/www/html/entrypoint.sh: no such file or directory": unknown

ERROR: for cms  Cannot start service cms: OCI runtime create failed: container_linux.go:344: starting container process caused "exec: \"/var/www/html/entrypoint.sh\": stat /var/www/html/entrypoint.sh: no such file or directory": unknown
ERROR: Encountered errors while bringing up the project.
```

I'm getting this even when running `docker-compose build cms`

**Comment by RoySegall** [comment]
The apache, composer and other stuff are set up correctly. Todos:
* [ ] Set the apache site to `web` directory so we won't need to access via `http://localhost:2000/webz` to Drupal
* [ ] Fix the issue with the admin/config
* [ ] When installing the new site, make sure the new media is set as the media button in CKEditor
* [ ] Handle GrpahQL requests
* [ ] Bonus - dummy data migration

**Comment by RoySegall** [comment]
I gave it another look, from what I've it's working but there are some issues:
1. The CSS is broken - I'll look it up.
2. Don't know how this will work with development - does any changes in the profiles folder will apply immediately? 
3. There's more sutff to do in https://github.com/hasadna/open_pension/pull/462#issuecomment-569452555

**Comment by RoySegall** [comment]
never mind CSS - issues with my machine. Keep checking this one up. Anyway, we have graphql:
image

**Comment by RoySegall** [comment]
@ItamarGronich I want to be sure - I can see changes in the code. That happens because of https://github.com/hasadna/open_pension/blob/56ab8908c2cd861b3b240f735443ec7af5de88cd/docker-compose.override.yml#L39, right?

**Comment by ItamarGronich** [comment]
> I gave it another look, from what I've it's working but there are some issues:
> 
> 1. The CSS is broken - I'll look it up.
> 2. Don't know how this will work with development - does any changes in the profiles folder will apply immediately?
> 3. There's more sutff to do in [#462 (comment)](https://github.com/hasadna/open_pension/pull/462#issuecomment-569452555)

1. you solved it
2. Profiles will update immediately. But.... composer won't. Let's live with this for a while and maybe solve this later on.
3. OK. Don't you want to Just merge this and start with a new PR? This one's big enough already.

**Comment by ItamarGronich** [comment]
> never mind CSS - issues with my machine. Keep checking this one up. Anyway, we have graphql:
> image

YES!

**Comment by ItamarGronich** [comment]
> @ItamarGronich I want to be sure - I can see changes in the code. That happens because of
> 
> https://github.com/hasadna/open_pension/blob/56ab8908c2cd861b3b240f735443ec7af5de88cd/docker-compose.override.yml#L39
> 
> , right?

Exactly.

**Comment by RoySegall** [comment]
> OK. Don't you want to Just merge this and start with a new PR? This one's big enough already.

Need to handle a couple of changes from yesterdat... but yes, i'll merge it.

**Comment by RoySegall** [comment]
@ItamarGronich The basic drupal is working. I'll open up a follow up issues for the other stuff with Drupal. For some reason travis is failing.

**Comment by RoySegall** [comment]
@ItamarGronich I need you to export the username and password credentials in travis to all branches:
image


It's preventing from this PR to connect to docker

**Comment by RoySegall** [comment]
@ItamarGronich Don't merge yet, there's a small feature I'm adding but we need to fix travis issues

**Comment by RoySegall** [comment]
@ItamarGronich now it's ready for merge - I managed to not re installing Drupal after it's already installed.

**Comment by RoySegall** [comment]
@ItamarGronich yay! the cms is connecting to the processr:
image

The next step is to connect it to the pipeline, but first - let's merge this one. it failing due to this one - https://github.com/hasadna/open_pension/pull/462#issuecomment-574803788

**Comment by RoySegall** [comment]
image
More work need to be done when triggering multiple files - Need to fix this little one thing and we can merge this one.

**Pull request merged**